android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -371,18 +371,22 @@ public void onLeScan(final BluetoothDevice device, final int rssi, final byte[]
           @Override
           public void run() {
             UriBeacon uriBeacon = UriBeacon.parseFromBytes(scanResult.getScanRecord().getBytes());
-            if ((uriBeacon != null) && (uriBeacon.getUriString() != null)) {
-              int txPower = uriBeacon.getTxPowerLevel();
+            if (uriBeacon != null) {
               String url = uriBeacon.getUriString();
-              // If we haven't yet seen this url
-              if (!mUrlToUrlMetadata.containsKey(url)) {
-                mUrlToUrlMetadata.put(url, null);
-                mNearbyDeviceAdapter.addItem(url, scanResult.getDevice().getAddress(), txPower);
-                // Fetch the metadata for this url
-                MetadataResolver.findUrlMetadata(getActivity(), NearbyBeaconsFragment.this, url, txPower, rssi);
+              if (url != null && !url.isEmpty()) {
+                int txPower = uriBeacon.getTxPowerLevel();
+                String address = scanResult.getDevice().getAddress();
+                // If we haven't yet seen this url
+                if (!mUrlToUrlMetadata.containsKey(url)) {
+                  mUrlToUrlMetadata.put(url, null);
+                  mNearbyDeviceAdapter.addItem(url, address, txPower);
+                  // Fetch the metadata for this url
+                  MetadataResolver.findUrlMetadata(getActivity(), NearbyBeaconsFragment.this, url,
+                                                   txPower, rssi);
+                }
+                // Tell the adapter to update stored data for this url
+                mNearbyDeviceAdapter.updateItem(url, address, scanResult.getRssi(), txPower);
```

Comment:
Here you use `scanResult.getRssi` but in `findUrlMetadata` you use the `rssi` passed to the function.